bundler/lib/bundler/templates/newgem/newgem.gemspec.tt

```diff
@@ -22,7 +24,7 @@ Gem::Specification.new do |spec|
 
   # Specify which files should be added to the gem when it is released.
   # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
-  spec.files         = Dir.chdir(File.expand_path("..", __FILE__)) do
+  spec.files         = Dir.chdir(File.expand_path(__dir__)) do
```

Comment:
The remaining 2 offenses that I mentioned in the description comes from here:
```
Offenses:

foo.gemspec:22:13: C: Layout/ExtraSpacing: Unnecessary spacing detected.
  spec.files         = Dir.chdir(File.expand_path(__dir__)) do
            ^^^^^^^^
foo.gemspec:22:22: C: Layout/SpaceAroundOperators: Operator = should be surrounded by a single space.
  spec.files         = Dir.chdir(File.expand_path(__dir__)) do
                     ^

8 files inspected, 2 offenses detected
```

I don't know what to do with it, because it doesn't look nice if we remove those spaces.
The one possible option here is to add the following lines to `.rubocop.yml`:
```
Layout/ExtraSpacing:
  Enabled: true
  Exclude:
    - newgem.gemspec.tt

Layout/SpaceAroundOperators:
  Enabled: true
  Exclude:
    - newgem.gemspec.tt
```

But I am not sure how right would it be to do this.

Reply:
I personally dislike the spaces, so I would remove then.

Reply:
Ah, sure! I'd love that myself, but it would be inconsistent with other things in the gemspec:
```
--- a/foo.gemspec
+++ b/foo.gemspec
@@ -19,7 +19,7 @@ Gem::Specification.new do |spec|
 
   # Specify which files should be added to the gem when it is released.
   # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
-  spec.files         = Dir.chdir(File.expand_path(__dir__)) do
+  spec.files = Dir.chdir(File.expand_path(__dir__)) do
     `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
   end
   spec.bindir        = "exe"
```

Notice the spaces b/w `spec.bindir` and `= "exe"`. Similarly, the entire gemspec is nicely formatted and `spec.files` would be odd one out, I feel.

Personally, I don't mind doing this, but it's just that I'd want to let you know this so we don't have any unexpected surprises :smile:

Reply:
Oh, kill them all! I believe `rubocop` should actually start complaining about the `spec.bindir` line as soon as you remove the spaces on the `spec.files` line.

Reply:
Aye, aye, cap'n!
I'll do this and also fix the specs :sweat_smile:

Reply:
> Oh, kill them all! I believe `rubocop` should actually start complaining about the `spec.bindir` line as soon as you remove the spaces on the `spec.files` line.

Just btw, `rubocop` is happy just after removing the spaces from `spec.files` :P
It doesn't care about anything else :joy:

Reply:
Oh, I see, it allows those spaces "for alignment" with the other assignments. So to completely fix this you need to either remove spaces in all assignments, or change the `spec.files` into a one-liner, so that `rubocop` detects that the spaces are there for alignment.

I don't care much about which solution you choose, so go ahead with the one you prefer.

Reply:
Ah, this is interesting! I found out why:
```
--- a/foo.gemspec
+++ b/foo.gemspec
@@ -9,6 +9,10 @@ Gem::Specification.new do |spec|
   spec.summary       = "TODO: Write a short summary, because RubyGems requires one."
   spec.description   = "TODO: Write a longer description or delete this line."
   spec.homepage      = "TODO: Put your gem's website or public repo URL here."
+  spec.files         = Dir.chdir(File.expand_path(__dir__)) do
+    `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
+  end
+
   spec.required_ruby_version = Gem::Requirement.new(">= 2.3.0")
 
   spec.metadata["allowed_push_host"] = "TODO: Set to 'http://mygemserver.com'"
```

If the spaces are being done in a block of lines, then it doesn't care.
`rubocop` registers no offense here and is happy!
On the other hand, this:
```
  # Specify which files should be added to the gem when it is released.
  # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
  spec.files         = Dir.chdir(File.expand_path(__dir__)) do
    `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
  end
  spec.bindir        = "exe"
  spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
  spec.require_paths = ["lib"]
```
is being registered as an offense. Interesting!